File: johnmatheus/wesley_safadao.go
package main

import (
	"flag"
	"fmt"
	"time"
)

func somatorio(x int) int {
	total := 0
	for ; x > 0; x-- {
		total += x
	}
	return total
}

func main() {

	data := flag.String("data", "00-00-0000", "Data de aniversário")
	flag.Parse()

	newdata, _ := time.Parse("02-01-2006", *data)

	safadeza := somatorio(int(newdata.Month())) + (newdata.Year()/1000)*(50-newdata.Day())
	anjo := 100 - safadeza
	fmt.Println("Safadeza: ", safadeza)
	fmt.Println("Anjo :", anjo)
}

File: johnmatheus/wesley_safadao.py
import datetime

data = input("Entre com sua data de nascimento: DD-MM-YYYY: ")
data = datetime.datetime.strptime( data, "%d-%m-%Y" )

ano = data.year
mes = data.month
dia = data.day

safadeza = sum( [i for i in range(1, int(mes)+1)] ) + (int(ano)/1000) * (50 - int(dia))
anjo = 100 - safadeza

print("Safadeza: {:.2f}\nAnjo: {:.2f}".format(safadeza, anjo), )
